src/worker.js

```diff
@@ -197,23 +222,36 @@ this.onmessage = function(e) {
 #if STACK_OVERFLOW_CHECK
         Module['checkStackCookie']();
 #endif
+#if !MINIMAL_RUNTIME // In MINIMAL_RUNTIME the noExitRuntime concept does not apply to pthreads. To exit a pthread with live runtime, use the function emscripten_unwind_to_js_event_loop() in the pthread body.
```

Comment:
this sort of difference seems dangerously confusing. is it well documented? Are there assertions that can catch the corner cases?

Reply:
Yes, there should be clear diagnostics messages that were added in #10174.